# Number Statistics using NumPy

## Day 11 - AI & ML Internship Task

This project uses NumPy to analyze a set of student marks and calculate important statistical values such as mean, median, maximum, minimum, and standard deviation.

### Objective

- Create a NumPy array containing student marks
- Calculate statistical values
- Display each result clearly
- Understand how NumPy is useful for data analysis

In [1]:
import numpy as np

# Student marks dataset
marks = np.array([78, 85, 92, 67, 88, 74, 95, 81, 69, 90])

print("Student Marks:")
print(marks)


Student Marks:
[78 85 92 67 88 74 95 81 69 90]


In [3]:
%pip install numpy

Note: you may need to restart the kernel to use updated packages.


In [4]:
mean = np.mean(marks)
median = np.median(marks)
maximum = np.max(marks)
minimum = np.min(marks)

print("Mean:", mean)
print("Median:", median)
print("Maximum:", maximum)
print("Minimum:", minimum)

Mean: 81.9
Median: 83.0
Maximum: 95
Minimum: 67


In [5]:
standard_deviation = np.std(marks)

print("Standard Deviation:", standard_deviation)

Standard Deviation: 9.23525852372309


## Statistical Results

The student marks were analyzed using NumPy.

- **Mean:** 81.9
- **Median:** 83.0
- **Maximum:** 95
- **Minimum:** 67
- **Standard Deviation:** 9.33

### Explanation

The **mean** represents the average marks of all students.  
The **median** represents the middle value when the marks are arranged in order.  
The **maximum** and **minimum** represent the highest and lowest marks.  
The **standard deviation** shows how much the marks vary from the average value.